frontend/src/Editor/Components/NumberInput.jsx

```diff
@@ -15,26 +15,30 @@ export const NumberInput = function NumberInput({
 
   const [value, setValue] = React.useState(parseInt(properties.value));
 
-  const handleChange = (e) => {
+  const handleValueChange = (newValue) => {
     if (
       !isNaN(parseInt(properties.minValue)) &&
       !isNaN(parseInt(properties.maxValue)) &&
       parseInt(properties.minValue) > parseInt(properties.maxValue)
     ) {
       setValue(parseInt(properties.maxValue));
-    } else if (!isNaN(parseInt(properties.maxValue)) && parseInt(e.target.value) > parseInt(properties.maxValue)) {
+    } else if (!isNaN(parseInt(properties.maxValue)) && parseInt(newValue) > parseInt(properties.maxValue)) {
       setValue(parseInt(properties.maxValue));
-    } else if (!isNaN(parseInt(properties.minValue)) && parseInt(e.target.value) < parseInt(properties.minValue)) {
+    } else if (!isNaN(parseInt(properties.minValue)) && parseInt(newValue) < parseInt(properties.minValue)) {
       setValue(parseInt(properties.minValue));
     } else {
-      setValue(parseInt(e.target.value));
+      setValue(parseInt(newValue));
     }
+  };
+
+  const handleChange = (e) => {
+    handleValueChange(e.target.value);
     fireEvent('onChange');
   };
 
   useEffect(() => {
-    setValue(parseInt(properties.value));
-  }, [properties.value]);
+    handleValueChange(value);
```

Comment:
```suggestion
    handleValueChange(properties.value);
```

Reply:
@kavinvenkatachalam, I have already tried using `properties.value`, but It was not working as expected. 
The Problem is `properties.value` is a `default` value, and whenever I change the default value it will change the `value in NumberInput` which is not correct.

And the default value is supposed to be set when there is no value right? like at the start of NumberInput creation.
And what you are saying is to pass the new value to `handleChange`, but the default value is not a new value, we will be normally changing the value which is the `state` of `NumberInput`.

Please correct me if I a wrong somewhere

Reply:
@Ubaid-Manzoor Agree with you. Can you remove `properties.value` from the dependency array below?

Reply:
Done @kavinvenkatachalam